src/Components/CreateImageWizard/steps/Oscap/components/PolicySelector.tsx

```diff
@@ -0,0 +1,314 @@
+      description={descr}
+    >
+      {policy.title}
+    </SelectOption>
+  );
+};
+
+const PolicySelector = () => {
+  const policyID = useAppSelector(selectCompliancePolicyID);
+  const policyTitle = useAppSelector(selectCompliancePolicyTitle);
+  const release = removeBetaFromRelease(useAppSelector(selectDistribution));
+  const majorVersion = release.split('-')[1];
+  const hasWslTargetOnly = useHasSpecificTargetOnly('wsl');
+  const dispatch = useAppDispatch();
+  const [isOpen, setIsOpen] = useState(false);
+  const [selected, setSelected] = useState<string>('None');
+
+  const {
+    data: policies,
+    isFetching: isFetchingPolicies,
+    isSuccess: isSuccessPolicies,
+  } = usePoliciesQuery({
+    filter: `os_major_version=${majorVersion}`,
+  });
+
+  const { data: currentProfileData } = useGetOscapCustomizationsForPolicyQuery(
+    {
+      distribution: release,
+      policy: policyID!,
+    },
+    { skip: !policyID }
+  );
+
+  const [trigger] = useLazyGetOscapCustomizationsForPolicyQuery();
+
+  useEffect(() => {
+    if (!policies || policies.data === undefined) {
+      return;
+    }
+
+    if (policyID && !policyTitle) {
+      for (const p of policies.data) {
+        const pol = p as PolicyRead;
+        if (pol.id === policyID) {
+          dispatch(
+            changeCompliance({
+              policyID: pol.id,
+              profileID: pol.ref_id,
+              policyTitle: pol.title,
+            })
+          );
+        }
+      }
+    }
+  }, [isSuccessPolicies]);
+
+  const handleToggle = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const handleClear = () => {
+    dispatch(
+      changeCompliance({
+        profileID: undefined,
+        policyID: undefined,
+        policyTitle: undefined,
+      })
+    );
+    clearOscapPackages(currentProfileData?.packages || []);
+    dispatch(changeFileSystemConfigurationType('automatic'));
+    handleServices(undefined);
+    dispatch(clearKernelAppend());
+  };
+
+  const handlePackages = (
+    oldOscapPackages: string[],
+    newOscapPackages: string[]
+  ) => {
+    clearOscapPackages(oldOscapPackages);
+
+    for (const pkg of newOscapPackages) {
+      dispatch(
+        addPackage({
+          name: pkg,
+          summary: 'Required by chosen compliance policy',
+          repository: 'distro',
+        })
+      );
+    }
+  };
+
+  const clearOscapPackages = (oscapPackages: string[]) => {
+    for (const pkg of oscapPackages) {
+      dispatch(removePackage(pkg));
+    }
+  };
+
+  const handlePartitions = (oscapPartitions: Filesystem[]) => {
+    dispatch(clearPartitions());
+
+    const newPartitions = oscapPartitions.map((filesystem) => {
+      const [size, unit] = parseSizeUnit(filesystem.min_size);
+      const partition: Partition = {
+        mountpoint: filesystem.mountpoint,
+        min_size: size.toString(),
+        unit: unit as Units,
+        id: uuidv4(),
+      };
+      return partition;
+    });
+
+    if (newPartitions) {
```

Comment:
What is the purpose of this? Even if newPartitions is empty, this will go through, I think.
And I do not think that map can return anything else than an array, or is that possible?

Reply:
ah you're right, this might need fixing in both the profile and policy selector.